src/framework/theme/components/layout/layout.component.ts

```diff
@@ -134,32 +134,44 @@ export class NgaLayoutComponent implements OnDestroy, AfterViewInit {
     this.centerValue = convertToBoolProperty(val);
   }
 
+  protected searchSubscription: Subscription;
+
   @ViewChild('layoutTopDynamicArea', { read: ViewContainerRef }) veryTopRef: ViewContainerRef;
 
+
   protected themeSubscription: Subscription;
   protected appendSubscription: Subscription;
   protected clearSubscription: Subscription;
 
   constructor(protected themeService: NgaThemeService,
-              protected componentFactoryResolver: ComponentFactoryResolver,
-              protected elementRef: ElementRef,
-              protected renderer: Renderer2) {
+    protected componentFactoryResolver: ComponentFactoryResolver,
+    protected elementRef: ElementRef,
+    protected renderer: Renderer2) {
     this.themeSubscription = this.themeService.onThemeChange().subscribe((theme) => {
 
       if (theme.previous) {
         this.renderer.removeClass(this.elementRef.nativeElement, 'theme-' + theme.previous);
       }
       this.renderer.addClass(this.elementRef.nativeElement, 'theme-' + theme.name);
     });
+
+    this.searchSubscription = this.themeService.onSearchActivate().subscribe((searchState) => {
```

Comment:
This logic should not belong to the layout component and must be moved into the search component itself.